### DISCUSSION TITLE
Python-GitLab: error connecting to GitLab instance

### DISCUSSION THREAD

**Discussion by cesaregerbino** [python-gitlab/python-gitlab#2388] [Q&A] [answered]
I'm using a CentOS7 virtual machine created in Oracle VirtualBox.

I've installed on it python with `sudo yum install python3`and then python-gitlab using `sudo yum install python-gitlab`

Python version is: 3.6.8
Python-Gitab version is: latest version (at the date, just installed)

I've tried this python code

    import gitlab
    import os
       
    gl = gitlab.Gitlab(url='https://<my gitab instance>/', private_token='my_token')
            
    def get_projects():
        projects = gl.projects.list(owned=True)
        for project in projects:
            print(project.name)
       
    def main():
        get_projects()
        
    main()

but it doesn't wok and the error is

    Traceback (most recent call last):
      File "Test.py", line 1, in <module>
        import gitlab
      File "/media/sf_SharedWithVM/VM-Python/GitLab/gitlab.py", line 6, in <module>
    AttributeError: 'module' object has no attribute 'Gitlab'

Any suggestion will be appreciated. Thank you in advance

**Comment by nejch** [comment, marked as answer]
@cesaregerbino rename your script from `gitlab.py` to something else ;) the namespace clashes with the package, which is also `gitlab`.

**Reply by cesaregerbino** [reply]
Hi @neich!  I've just rename the script in Test.py but the problem remain ... :-(

**Reply by JohnVillalovos** [reply]
To help debug it you might try:

```
import gitlab
print(gitlab.__file__)
```

**Reply by nejch** [reply]
Yeah, I'd say you either still have a `gitlab.py` file or a directory named gitlab in your python path and there is a clash.

**Reply by cesaregerbino** [reply]
You're right @neich !!! My Test.py file was, unfortunately, in a "GitLab" directory !!!! Thanks a lot!
